src/auth/src/credentials/mds.rs

```diff
@@ -455,6 +463,44 @@ mod test {
     }
 
     #[tokio::test]
+    #[serial]
+    async fn test_gce_metadata_host_env_var() {
+        let scopes = ["scope1".to_string(), "scope2".to_string()];
+        let response = MDSTokenResponse {
+            access_token: "test-access-token".to_string(),
+            expires_in: Some(3600),
+            token_type: "test-token-type".to_string(),
+        };
+        let response_body = serde_json::to_value(&response).unwrap();
+
+        let (endpoint, _server) = start(Handlers::from([(
+            format!("{}/token", MDS_DEFAULT_URI),
+            (
+                StatusCode::OK,
+                response_body,
+                TokenQueryParams {
+                    scopes: Some(scopes.join(",")),
+                    recursive: None,
+                },
+                Arc::new(Mutex::new(0)),
+            ),
+        )]))
+        .await;
+
+        // Trim out 'http://' from the endpoint provided by the fake server
+        let _e = ScopedEnv::set(super::GCE_METADATA_HOST_ENV_VAR, &endpoint[7..]);
```

Comment:
Hmmm.... I think I would prefer:

```suggestion
        let _e = ScopedEnv::set(super::GCE_METADATA_HOST_ENV_VAR, endpoint.strip_prefix("http://").unwrap_or(endpoint));
```